.ado/publish.yml

```diff
@@ -0,0 +1,164 @@
+
+# TODO: sign binaries/packages
+# TODO: publish packages to Nuget.org and npmjs
+# TODO: publish symbols
+# TODO: publish to source server
+
+pr: none
+trigger:
+- main
+
+parameters:
+  # x86 versions are not supported by NativeAOT
+  # win-arm64 is not supported by Node.JS as of 3/29/2023
+  # linux-arm64 must be added after we learn how to set up the cross-compilation environment.
+  # Use net7.0 as the latest stable version except for osx-arm64 which is only supported in net8.0.
+- name: buildMatrix
+  type: object
+  default:
+    - Name: win_x64
+      VMImage: windows-latest
+      TargetRuntime: win-x64
+      DotNetVersion: '7.0.x'
+      DotNetMoniker: net7.0
+    - Name: osx_x64
+      VMImage: macos-latest
+      TargetRuntime: osx-x64
+      DotNetVersion: '7.0.x'
+      DotNetMoniker: net7.0
+    - Name: osx_arm64
+      VMImage: macos-latest
+      TargetRuntime: osx-arm64
+      DotNetVersion: '8.0.x'
+      DotNetMoniker: net8.0
+    - Name: linux_x64
+      VMImage: ubuntu-latest
+      TargetRuntime: linux-x64
+      DotNetVersion: '7.0.x'
+      DotNetMoniker: net7.0
+
+jobs:
+  - ${{ each matrixEntry in parameters.buildMatrix }}:
+    - job: buildAOTBinary${{ matrixEntry.Name }}
+      displayName: Build ${{ matrixEntry.TargetRuntime }} AOT binary
+      pool:
+        vmImage: ${{ matrixEntry.VMImage }}
+
+      steps:
+        - checkout: self
+          displayName: Deep git fetch for version generation
+          fetchDepth: 0 # Use deep fetch for the version calculation by Nerdbank.GitVersioning
+          clean: false
+          submodules: false
+          lfs: false
+
+        - task: UseDotNet@2
+          displayName: Install .Net
+          inputs:
+            packageType: 'sdk'
+            version: ${{ matrixEntry.DotNetVersion }}
+            includePreviewVersions: true
+
+        - task: DeleteFiles@1 # To enable net8.0 use for osx-arm64
+          displayName: Delete 'global.json'
+          inputs:
+            Contents: 'global.json'
+          condition: eq('${{ matrixEntry.DotNetMoniker }}', 'net8.0')
+
+        - script: dotnet publish --configuration Release --runtime ${{ matrixEntry.TargetRuntime }} --no-self-contained --framework ${{ matrixEntry.DotNetMoniker }}
+          displayName: Build native binaries
+          env:
+            TargetFrameworks: ${{ matrixEntry.DotNetMoniker }}
+
+        - task: CopyFiles@2
+          displayName: Copy build artifacts to staging
+          inputs:
+            sourceFolder: $(Build.SourcesDirectory)/out/bin/Release/NodeApi/${{ matrixEntry.DotNetMoniker }}/${{ matrixEntry.TargetRuntime }}/publish
+            targetFolder: $(Build.StagingDirectory)/AOT/${{ matrixEntry.TargetRuntime }}
+            contents: Microsoft.JavaScript.NodeApi.node
+
+        - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
+          displayName: "📒 Generate Manifest: ${{ matrixEntry.TargetRuntime }}"
```

Comment:
What does this task do? How is the manifest used?

Reply:
I copied it from the RNW pipeline. I assume it produces a manifest for all files that we upload as ADO artifacts.